app/src/main/java/com/example/courses/presentation/ui/MainFragment.kt

```diff
@@ -259,27 +262,43 @@ class MainFragment : Fragment(), ItemClickListener {
             addedPersonBroadcast,
             IntentFilter(Const.ADD_PERSON_ACTION)
         )
-        sensorManager.registerListener(acselerometorListener,sensor,SensorManager.SENSOR_DELAY_NORMAL)
+        sensorManager.registerListener(
+            acselerometorListener,
+            sensor,
+            SensorManager.SENSOR_DELAY_NORMAL
+        )
 
         if (ActivityCompat.checkSelfPermission(
                 requireContext(),
                 Manifest.permission.ACCESS_FINE_LOCATION
-            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+            ) != PackageManager.PERMISSION_GRANTED || ActivityCompat.checkSelfPermission(
```

Comment:
This makes check more restrictive. I'd suggest learning deeper regarding each location permission, separating scenarios before applying such check.